### DISCUSSION TITLE
[Support] 今天用云效打包到ECS时提示Property 'state' must be of type 'any', but here has type 'State'

### DISCUSSION THREAD

**Discussion by CWaiLeung** [midwayjs/midway#3479] [Q&A] [answered]
### Describe the problem（描述问题）

我在虚拟机Ubuntu 20.04.3 LTS中运行没有发现此问题。

上传到云效中打包时的提示：

node_modules/@midwayjs/web/index.d.ts:34:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'state' must be of type 'any', but here has type 'State'.

我用vscode打开相应文件，state是标红的

```
import {
  Context as IMidwayBaseContext,
  IMidwayBaseApplication,
} from '@midwayjs/core';
import {
  IMidwayWebBaseApplication,
  IMidwayWebConfigurationOptions,
  Context as EggContext,
  State,
} from './dist';
import { ILogger, LoggerOptions } from '@midwayjs/logger';
import { EggAppConfig } from 'egg';

export * from './dist/index';

declare module 'egg' {
  interface EggAppInfo {
    appDir: string;
  }

  // 这里再次覆盖和 egg 不同的定义，不然 egg 插件里可能会报错
  interface Application
    extends IMidwayBaseApplication<EggContext>,
      IMidwayWebBaseApplication {
    createAnonymousContext(...args: any[]): EggContext;
    getCoreLogger(): EggLogger & ILogger;
    getLogger(name?: string): EggLogger & ILogger;
    createLogger(name: string, options: LoggerOptions): EggLogger & ILogger;
  }

  interface Context<ResponseBodyT = any> extends IMidwayBaseContext {
    getLogger(name?: string): ILogger;
    forward: (url: string) => void;
    state: State;
  }
}

declare module '@midwayjs/core/dist/interface' {
  interface MidwayConfig extends PowerPartial<EggAppConfig> {
    egg?: IMidwayWebConfigurationOptions;
  }
}
```

=================================================================
旧文件内容

```
import {
  Context as IMidwayBaseContext,
  IMidwayBaseApplication,
} from '@midwayjs/core';
import {
  IMidwayWebBaseApplication,
  IMidwayWebConfigurationOptions,
  Context as EggContext,
} from './dist';
import { ILogger, LoggerOptions } from '@midwayjs/logger';
import { EggAppConfig } from 'egg';

export * from './dist/index';

declare module 'egg' {
  interface EggAppInfo {
    appDir: string;
  }

  // 这里再次覆盖和 egg 不同的定义，不然 egg 插件里可能会报错
  interface Application
    extends IMidwayBaseApplication<EggContext>,
      IMidwayWebBaseApplication {
    createAnonymousContext(...args: any[]): EggContext;
    getCoreLogger(): EggLogger & ILogger;
    getLogger(name?: string): EggLogger & ILogger;
    createLogger(name: string, options: LoggerOptions): EggLogger & ILogger;
  }

  interface Context<ResponseBodyT = any> extends IMidwayBaseContext {
    getLogger(name?: string): ILogger;
    forward: (url: string) => void;
  }
}

declare module '@midwayjs/core/dist/interface' {
  interface MidwayConfig extends PowerPartial<EggAppConfig> {
    egg?: IMidwayWebConfigurationOptions;
  }
}
```

我找回之前的已下载好的文件，发现state那一句是新加上去的，帮忙处理下，谢谢

### Midway Versions（Midway 版本）

>> Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.13.0)
✓ @midwayjs/axios(3.13.4)
✓ @midwayjs/bootstrap(3.13.0)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache(3.13.0)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(3.13.4)
✓ @midwayjs/core(3.13.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.13.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(3.13.4)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.13.4)
✓ @midwayjs/info(not installed)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.13.0)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(3.13.4)
✓ @midwayjs/redis(3.13.0)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(3.13.4)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.13.4)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.13.4)
✓ @midwayjs/version(3.13.4)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(3.13.4)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(not installed)
✓ @midwayjs/ws(3.13.4)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by CWaiLeung** [comment]
全部报错的内容

[10:20:32] #19 28.94 node_modules/@midwayjs/web/index.d.ts:34:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'state' must be of type 'any', but here has type 'State'.
[10:20:32] #19 28.94 
[10:20:32] #19 28.94 34     state: State;
[10:20:32] #19 28.94        ~~~~~
[10:20:32] #19 28.94 
[10:20:32] #19 28.94   node_modules/egg/index.d.ts:905:5
[10:20:32] #19 28.94     905     state: any;
[10:20:32] #19 28.94             ~~~~~
[10:20:32] #19 28.94     'state' was also declared here.
[10:20:32] #19 28.94 
[10:20:32] #19 28.98 
[10:20:32] #19 28.98 
[10:20:32] #19 28.98 
[10:20:33] #19 28.98 
[10:20:33] #19 28.98 Error! You can try adding the -V parameter for more information output.
[10:20:33] #19 28.98 
[10:20:33] #19 28.98 
[10:20:33] #19 28.98 
[10:20:33] #19 28.98 
[10:20:33] #19 29.02 Error: TS Error: Subsequent property declarations must have the same type.  Property 'state' must be of type 'any', but here has type 'State'.(node_modules/@midwayjs/web/index.d.ts)
[10:20:33] #19 29.02     at BuildPlugin.emit (/app/node_modules/@midwayjs/cli-plugin-build/src/index.ts:203:17)
[10:20:33] #19 29.02     at CommandCore.execLifecycle (/app/node_modules/@midwayjs/command-core/src/core.ts:216:11)
[10:20:33] #19 29.02     at CommandCore.invoke (/app/node_modules/@midwayjs/command-core/src/core.ts:182:5)
[10:20:33] #19 29.02     at CLI.start (/app/node_modules/@midwayjs/command-core/src/cli.ts:191:5)
[10:20:33] #19 ERROR: executor failed running [/bin/sh -c npm run build]: exit code: 1

**Comment by czy88840616** [comment, marked as answer]
务必在 tsconfig.json 中开启 "skipLibCheck": true

**Reply by CWaiLeung** [reply]
ok
